File: KITE-Example-Test/src/main/java/org/webrtc/kite/example/steps/GoogleSearchStep.java
package org.webrtc.kite.example.steps;

import io.cosmosoftware.kite.interfaces.Runner;
import io.cosmosoftware.kite.steps.TestStep;
import org.webrtc.kite.example.pages.GoogleSearchPage;

public class GoogleSearchStep extends TestStep {
  
  private final String TARGET = "CoSMo Software Consulting";
  private final GoogleSearchPage searchPage;
  
  public GoogleSearchStep(Runner runner) {
    super(runner);
    this.searchPage = new GoogleSearchPage(runner);
  }
  
  @Override
  protected void step() {
    
    searchPage.open();
//    searchPage.searchFor(TARGET);
  }
  
  @Override
  public String stepDescription() {
    return "Open " + GoogleSearchPage.getURL() + " and look for " + TARGET;
  }
}

File: KITE-Example-Test/js/steps/GoogleSearchStep.js
const {TestStep} = require('kite-common');
/**
 * Class: GoogleSearchStep
 * Extends: TestStep
 * Description:
 */
class GoogleSearchStep extends TestStep {
  constructor(kiteBaseTest, url) {
    super();
    this.driver = kiteBaseTest.driver;
    this.timeout = kiteBaseTest.timeout;
    this.url = url;
    this.page = kiteBaseTest.page;
    this.target = "CoSMo Software Consulting";
  }

  stepDescription() {
    return "Open " + this.url + " and look for " + this.target;
  }

  async step() {
    await this.page.open(this);
    await this.page.searchFor(this.target);
  }
}

module.exports = GoogleSearchStep;